temporian/io/test/polars_test.py

```diff
@@ -0,0 +1,496 @@
+
+        expected_evset = event_set(
+            timestamps=[1640995200, 1641081600, 1641168000],
+            features={
+                "product_id": [666964, 666964, 574016],
+                "costs": [740.0, 508.0, 573.0],
+            },
+            indexes=["product_id"],
+            is_unix_timestamp=True,
+        )
+
+        evset = from_polars(df, indexes=["product_id"], timestamps="timestamp")
+        self.assertEqual(evset, expected_evset)
+        self.assertTrue(evset.schema.is_unix_timestamp)
+
+    def test_invalid_boolean_timestamp_type(self) -> None:
+        df = pl.DataFrame(
+            {
+                "product_id": [666964, 666964, 574016],
+                "timestamp": [True, False, True],
+                "costs": [740.0, 508.0, 573.0],
+            }
+        )
+
+        with self.assertRaises(ValueError):
+            from_polars(df, indexes=["product_id"], timestamps="timestamp")
+
+    def test_invalid_string_timestamp_type(self) -> None:
+        df = pl.DataFrame(
+            {
+                "product_id": [666964, 666964, 574016],
+                "timestamp": ["A", "B", 1.0],
+                "costs": [740.0, 508.0, 573.0],
+            }
+        )
+
+        with self.assertRaises(ValueError):
+            from_polars(df, indexes=["product_id"], timestamps="timestamp")
+
+    def test_timestamps_with_missing_values(self) -> None:
+        df = pl.DataFrame(
+            {
+                "product_id": [666964, 666964, 574016, 574016],
+                "timestamp": [1.0, 2.0, 3.0, None],
+                "costs": [740.0, 508.0, 573.0, 573.0],
+            }
+        )
+
+        with self.assertRaises(ValueError):
+            from_polars(df, indexes=["product_id"], timestamps="timestamp")
+
+    def test_timestamps_with_non_supported_object(self) -> None:
+        with self.assertRaises(TypeError):
+            df = pl.DataFrame(
+                {
+                    "product_id": [666964, 666964, 574016, 574016],
+                    "timestamp": [1.0, 2.0, 3.0, object()],
+                    "costs": [740.0, 508.0, 573.0, 573.0],
+                }
+            )
+
+    def test_no_index(self) -> None:
+        df = pl.DataFrame(
+            {
+                "product_id": [666964, 666964, 574016],
+                "timestamp": [1.0, 2.0, 3.0],
+                "costs": [740.0, 508.0, 573.0],
+            }
+        )
+        expected_evset = event_set(
+            timestamps=[1.0, 2.0, 3.0],
+            features={
+                "product_id": [666964, 666964, 574016],
+                "costs": [740.0, 508.0, 573.0],
+            },
+        )
+        evset = from_polars(df, indexes=[], timestamps="timestamp")
+        self.assertEqual(evset, expected_evset)
+
+    def test_datetime_in_feature_column(self) -> None:
+        df = pl.DataFrame(
+            {
+                "product_id": [666964, 666964, 574016],
+                "costs": ["2022-01-01", "2022-01-02", "2022-01-03"],
```

Comment:
these are strings and will be treated as so by both polars and temporian, create them as dates manually for this test to do what its title says